test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageBuilderTest.cs

```diff
@@ -371,6 +374,42 @@ public void CreatePackageWithNuspecIncludeExcludeWithWildcards(string source, st
             }
         }
 
+        [PlatformTheory(Platform.Windows)]
+        [InlineData(@"dir1\dir2\**", true, "Content")]
+        [InlineData("dir1/dir2/**", false, "Content")]
+        public void PackageBuilder_AddFiles_HasDifferentBehaviorDependingOnSlash(string source, bool expectFlattened, string destination)
+        {
+            // https://github.com/NuGet/Home/issues/11234
+            using (var directory = new TestSourcesDirectory())
+            {
+                // Arrange
+                PackageBuilder builder = new PackageBuilder();
+
+                // Act
+                builder.AddFiles(directory.Path, source, destination);
+
+                // Assert
+                var expectedResults = new[]
+                {
+                    new
+                    {
+                        Path = expectFlattened ? $"Content{Path.DirectorySeparatorChar}file1.txt" : $"Content{Path.DirectorySeparatorChar}dir1{Path.DirectorySeparatorChar}dir2{Path.DirectorySeparatorChar}file1.txt",
```

Comment:
Isn't this `else` dependent on knowing the input to the test? That seems like a bad practice.

Reply:
Yeah its how I designed the test, I pass in a path with `/` and expect the files to be flattened and then paths with `\` and expect them to not be flattened.  I could do two separate tests but I foresee us fixing this "bug" and making them return the same things so this test would be updated to remove these ternary operators.  Do you think there's a better way to accomplish this?

Reply:
Maybe I'm not fully thinking about it, but I would think you'd pass in both the unflattened string and the flattened string to the test, that way you assert the 2 test inputs match after running through the method.

Reply:
I'm going to merge it as-is for now, but I will update it later.  It used to need to run on all platforms which is why I tried to make it so dynamic.  But it ended up only needing to be run on Windows so it could be made to have a constant value for `expected`.  I'll just make that change when this whole feature is fixed.